### ISSUE TITLE
after upgrade no new pastes

### ISSUE THREAD

**Issue by igregma** [PrivateBin/PrivateBin#90]
hi :-)

after an upgrade from the latest stable release 0.22 to the current branch I can't create new pastes anymore.
error (german inst):  Konnte Text nicht erstellen: Fehler auf dem Server oder keine Antwort vom Server
downgrade to 0.22 and everything is fine again.

maybe helpful:
If I don't delete all the stuff and just overwrite all files to upgrade "0.22 -> current" the displayed error switchs to:
 Konnte Text nicht erstellen: There is no suitable CSPRNG installed on your system

The System is Ubuntu 14-04, PHP 5.5.9-1

**Comment by elrido** [comment]
Hello igregma

With your PHP version 5.5, you need to provide/install one of the following sources of cryptographically safe pseudo random numbers (CSPRN, the G in CSPRNG would be the generator), as stated in the installation [requirements](https://github.com/PrivateBin/PrivateBin/wiki/Installation#requirements):
- [Libsodium](https://download.libsodium.org)
- open_basedir access to `/dev/urandom`
- mcrypt extension

The com_dotnet extension is Windows only and PHP 7 would require you to upgrade to Ubuntu 16.04 (you surely will do that eventually, but probably want a short term solution for now) or use a backport of PHP 7 for 14.04.

**Comment by rugk** [comment]
Here is a guide for installing libsodium: https://paragonie.com/book/pecl-libsodium/read/00-intro.md#installing-libsodium
AFAIK it should be quite easy.

**Comment by rugk** [comment]
Please also run [this test script](https://github.com/PrivateBin/PrivateBin/issues/80#issuecomment-24032447) if the previous instructions did not help.

**Comment by igregma** [comment]
Hello,

The fastest solution was to add unrandom to my php openbasedir settings,
everything works perfect now!
Thanks :-)

Am Mittwoch, 24. August 2016 schrieb El RIDO :

**Comment by rugk** [comment]
I also wrote a FAQ about this: https://github.com/PrivateBin/PrivateBin/wiki/FAQ

I also added some other nice questions. As it is a wiki everyone is welcome to edit it.

**Comment by rugk** [comment]
@igregma Any new issue?

**Comment by igregma** [comment]
Well... hello again :-)

After the next Upgrade - 0.22 to 1.0 - the error ist back?!

 _Konnte Text nicht erstellen: Fehler auf dem Server oder keine Antwort vom Server_

PHP version used: 5.5.9-1ubuntu4.19
extension libsodium is loaded: false
/dev/urandom exists and is readable: true
extension mcrypt is loaded: false
function mcrypt_create_iv exists: false
extension com_dotnet is loaded: false

**Comment by elrido** [comment]
What do you find in the webservers error log? Any stack trace shown there?

**Comment by rugk** [comment]
As you have one line, which evaluates to true (`/dev/urandom exists and is readable`) this should be used.

This might be an upstream issue with the lib we use. So can you please run this small script:
```php
<?php
            $RandomCompat_basedir = ini_get('open_basedir');
            if (!empty($RandomCompat_basedir)) {
                $RandomCompat_open_basedir = explode(
                    PATH_SEPARATOR,
                    strtolower($RandomCompat_basedir)
                );
                $RandomCompatUrandom = (array() !== array_intersect(
                    array('/dev', '/dev/', '/dev/urandom'),
                    $RandomCompat_open_basedir
                ));
                $RandomCompat_open_basedir = null;
            }

           var_dump($RandomCompat_basedir);
           var_dump($RandomCompat_open_basedir);
?>

([source](https://github.com/paragonie/random_compat/blob/master/lib/random.php#L88-L100))

**Comment by igregma** [comment]
string(46) "/var/www/vhosts/xxxxxx.de/:/tmp/:/dev/urandom" NULL

**Comment by rugk** [comment]
I have adjusted the script above.

**Comment by igregma** [comment]
string(46) "/var/www/vhostsxxxxx.de/:/tmp/:/dev/urandom" NULL bool(true)

**Comment by rugk** [comment]
Okay, so this is all right. Could you look for a stack trace as @elrido said?

**Comment by igregma** [comment]
Yes, but will be later this evening... disabled most of the webserver logging stuff

**Comment by elrido** [comment]
I don't think this is the problem this time. As the "Fehler auf dem Server oder keine Antwort vom Server" message is the generic error message that gets displayed, there was an internal server error. This means that there was an exception thrown or some kind of error that wasn't catched.

But it should show up in the error logs of your webserver / PHP process. Let's wait on these.

**Comment by igregma** [comment]
Weird.
I saw a php parse error because of the lib/data/filesystem.php

The file was only 1 kb, so I just downloaded it from the repository and everything works like charm!
I already deleted the relases-zip-file, so can't say if it was a unpack error or something like this...

Thanks for the help, keep up the great work guys :-)
